pkg/statelocker/statelocker_test.go

```diff
@@ -0,0 +1,317 @@
+package statelocker
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/google/uuid"
+
+	"github.com/openshift/node-observability-agent/pkg/runs"
+)
+
+const (
+	validUID string = "dd37122b-daaf-4d75-9250-c0747e9c5c47"
+)
+
+func TestLock(t *testing.T) {
+	testCases := []struct {
+		name           string
+		previousState  State
+		expectedState  State
+		errFileContent string
+		expectedError  bool
+	}{
+		{
+			name:          "turnTaker was free, can take turn",
+			previousState: Free,
+			expectedState: Free,
+			expectedError: false,
+		},
+		{
+			name:          "turnTaker was taken, returns the id of running job",
+			previousState: Taken,
+			expectedState: Taken,
+			expectedError: false,
+		},
+		{
+			name:           "turnTaker was in error, returns the id of job in error",
+			previousState:  InError,
+			expectedState:  InError,
+			errFileContent: "{\"ID\":\"" + validUID + "\",\"ProfilingRuns\":[{\"Type\":\"Kubelet\",\"Successful\":false,\"BeginDate\":\"2022-03-03T10:10:17.188097819Z\",\"EndDate\":\"2022-03-03T10:10:47.211572681Z\",\"Error\":\"fake error\"},{\"Type\":\"CRIO\",\"Successful\":true,\"BeginDate\":\"2022-03-03T10:10:17.188499431Z\",\"EndDate\":\"2022-03-03T10:10:47.215840909Z\",\"Error\":null}]}",
+			expectedError:  false,
+		},
+		{
+			name:           "turnTaker was in error, and unable to read error file, error",
+			previousState:  InError,
+			expectedState:  InError,
+			errFileContent: "{\"ID" + validUID + "\",\"ProfilingRuns\":{\"Type\":\"Kubelet\",\"Successful\":false,\"BeginDate\":\"2022-03-03T10:10:17.188097819Z\",\"EndDate\":\"2022-03-03T10:10:47.211572681Z\",\"Error\":\"fake error\"},{\"Type\":\"CRIO\",\"Successful\":true,\"BeginDate\":\"2022-03-03T10:10:17.188499431Z\",\"EndDate\":\"2022-03-03T10:10:47.215840909Z\",\"Error\":null}]}",
+			expectedError:  true,
+		},
+	}
+	for i, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			var previousJobID uuid.UUID
+			errFileName := "/tmp/agent_tt" + fmt.Sprint(i) + ".err"
+			turnTaker := NewStateLock(errFileName)
+			if tC.previousState == InError {
+				previousJobID = uuid.MustParse(validUID)
+				// prepare an error file
+				errorFile := errFileName
+				err := os.WriteFile(errorFile, []byte(tC.errFileContent), 0644)
```

Comment:
```suggestion
			turnTaker := NewStateLock(errFileName)
			if tC.previousState == InError {
				previousJobID = uuid.MustParse(validUID)
				// prepare an error file
				err := os.WriteFile("/tmp/agent_tt" + fmt.Sprint(i) + ".err", []byte(tC.errFileContent), 0644)
```

Reply:
But we also would need the file name for `stateLock := NewStateLock(errFileName)`

Reply:
Yes, sorry, my bad, missed this one.